### PR TITLE
Change the config handling

### PR DESCRIPTION
**Pull request by tomdele**
Better paths handling in config
* Use pathlib instead of os.path
* manifest not mandatory anymore
* Better handling of the relative paths in manifest and in the different paths
   - can use all the "." + something (i.e: ./dir, ../, ./../, etc)
* strings resolved as path only if contains '$' or starts with '.'
* Add raises to avoid errors :
   - double abs path resulting in single abs path
   - manifest anchor with more than one sub-anchor
   - check if all the paths returned are absolute paths (when possible)
   - if an anchor (which is an abspath) is badly placed

### CODE REVIEW

**Comment by asanin-epfl** [comment]
@jonathanlurie I will go after you here.

**Comment by tomdele** [comment]
Rah need to fix the python2 first

File: bluepysnap/config.py [hunk 1]
```diff
@@ -56,33 +57,46 @@ def _resolve_manifest(manifest, configdir):
...
-                result[k] = os.path.join(configdir, v[2:])
+            if v.startswith('.'):
+                result[k] = str(Path(configdir, v).resolve())
 
         while True:
             update = False
             for k, v in six.iteritems(result):
+                if v.count('$') > 1:
+                    raise BluepySnapError(
+                        '{} is not a valid anchor : contains more than one sub anchor.')
```
Comment (joni-herttuainen) [inline bluepysnap/config.py:68]: missing `.format()`

Comment (tomdele) [reply bluepysnap/config.py:68]: indeed !

**Comment by tomdele** [comment]
Thank you guys !

**Pull request merged**